specs/vep-1243-vdk-meta-jobs/README.md

```diff
@@ -153,6 +188,107 @@ Consider at least the below topics but you do not need to cover those that are n
+
+Generally, job1 is independent of job2 in terms of requests - job1 would not impact the number of requests job2 would
+send throughout its duration.
+
+Thus, 1 job would send (1 + E/S + 1) API request for its run.
+Hence, the DAG would make J * (1 + E/S + 1) total requests.
+
+For example, the following workflow consists of 6 Data Jobs:
+
+* `Job 2`, `Job 3` and `Job 4` depend on `Job 1`;
+* `Job 5` depends on `Job 2` and `Job 3`;
+* `Job 6` depends on `Job 4`;
+* The execution time of each of the following jobs is about 10 min: `Job 1` and `Job 2`;
+* The execution time of each of the following jobs is about 15 min: `Job 3`, `Job 4` and `Job 6`;
+* The execution time of each of the following jobs is about 20 min: `Job 5`;
+* The concurrent running jobs limit is configured to 2.
+* The value of META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS is configured to the default 40 (seconds).
+
+![sample-execution.png](sample-execution.png)
+
+In this scenario, the vdk-meta-jobs plugin will perform a total of 147 requests for the whole workflow execution.
+See Appendix A for more details about this example.
+
+### Troubleshooting
+
+* What are the possible failure modes.
+
+  * There are a few ways in which the plugin may fail:
+
+    * If an orchestrated job is being finalized and its status is not the expected "succeeded" one and the configuration
+    value of "fail_meta_job_on_error" is set to TRUE, then a USER ERROR is raised for the Meta Job and its execution
+    is interrupted.
+    * If any kind of platform error occurs, a Meta Job would retry up to N (configurable by the Control Service) times
+      for each job it is orchestrating.
+    * If a non-registered job is attempted to be accessed, the plugin would raise Index Error with some suggestions.
+    * If a job is starting but a conflict is detected with another running job (409) or an internal server error (>=500)
+      has occurred, the execution of the job would be delayed.
+    * In the event of circular dependency (e.g. job1 depends on job2 and job2 depends on job1), violating the
+      requirements, the plugin would raise User Error.
+
+  * Diagnostics: All the main operations from the creation of the Meta Jobs DAG to the very end of the execution are
+    logged.
+
+  * Testing: There are [tests](/projects/vdk-plugins/vdk-meta-jobs/tests/test_meta_job.py) for all the aforementioned
+    failure scenarios.
+
+### Test Plan
+
+Unit and functional tests are introduced as part of the VDK Meta Jobs plugin.
+
+### Availability
+
+The plugin will be part of the vdk installation, so the same availability constraints apply.
+
+### Dependencies
+
+There are no external dependencies that the plugin relies on.
+
+### Cons
+
 Although having all these pros, the user should have in mind the cons of the Meta Jobs as well. The limitation that
 have to be considered is that there is a limit on the number of concurrent running jobs in a single DAG. If this
```

Comment:
`the number of parallel running jobs`

Reply:
I started thinking about concurrency and parallelism in this context and, honestly, I got a bit confused about it. I am not entirely sure both terms are OK (they probably are) to describe this scenario.

Reply:
Alright, makes sense. Will change this.